### ISSUE TITLE
Cannot update single nested subdocument.

### ISSUE THREAD

**Issue by EmailDE** [Automattic/mongoose#6152]
I want to update a single nested subdocument from its own 'save' middleware. It is just more convenient than doing it on its parent schema. Are there plans for this feature? Is this a bug? I hope there is a neat solution for this. I have also provided my approach below. I am using version 5.0.6 of mongoose. Thanks in advance :)

```
const dotenv = require('dotenv');
const mongoose = require('mongoose');

dotenv.config();

mongoose.connect(process.env.DATABASEURI, function(err) {
	if(err) throw err;
	console.log("Established connection to database \"" + process.env.DATABASEURI + "\"");
});

const Schema = mongoose.Schema;

const childSchema = new Schema({ name: String, age: Number });

childSchema.pre('save', function(next) {
	this.update({ $inc: { age: 1 } }, next);
});

const parentSchema = new Schema({
  	name: String,
  	child: childSchema
});

const ParentModel = mongoose.model('test', parentSchema);

ParentModel.create({
	name: 'x',
	child: {
		name:'y'
	}
}, function(err, doc) {
	if(err) throw err;

	console.log(doc);
	mongoose.connection.close();
});
```

**Comment by vkarpov15** [comment]
Why don't you just do `++this.age` in the `pre('save')`? If you're in pre('save') mongoose is already going to save the document anyway, why do you need to do an extra update?

**Comment by EmailDE** [comment]
This was just a concept. Maybe I should have provided a better example which made it more clear why I need to update the document. Lets assume that I want to increment the login attempts of a user and that I have more the one node application accessing the database. Hopefully now it is clear why I need to "update" the document rather than just "saving" it. There is a subschema handling the login and a schema handling the whole user. It would just be nice to be able to call the update from the subschema and its middlewares.

**Comment by vkarpov15** [comment]
Yeah this is a tricky one. You currently would have to do something like this:

```javascript
childSchema.pre('save', function(next) {
	if(this.isNew) return next();

        var path = this.$basePath + '.age';
	this.ownerDocument().update({ $inc: { [path]: 1 } }, next);
});
```

Because we don't currently support doing `update()` on nested docs. This is definitely something we would like to add though, thanks for the suggestion.

**Comment by EmailDE** [comment]
Thanks for your answer. I am definitely looking forward to this feature.

**Comment by EmailDE** [comment]
Your solution also works with diabled "_id" field on the subdocument. I am just hoping this does not change with the feature.

**Comment by EmailDE** [comment]
I did some testing on the behaviour of the "save" function. It also does not work on single nested subdocuments, yet. The difference is that it fails silently. Even the document in the callback contains the changes, but they are not persistent in the database. I modified my previous example a bit. Obviously you could call "doc.save", but - again - being able to call "save" on the subdocument comes in handy when using the subdocument's middleware. I just wanted to point it out. I do not need assistance with this one ^^.

```
const dotenv = require('dotenv');
const mongoose = require('mongoose');

dotenv.config();

mongoose.set('debug', true);

mongoose.connect(process.env.DATABASEURI, function(err) {
	if(err) throw err;
	console.log("Established connection to database \"" + process.env.DATABASEURI + "\"");
});

const Schema = mongoose.Schema;

const childSchema = new Schema({ name: String });

const parentSchema = new Schema({
  	name: String,
  	child: childSchema
});

const ParentModel = mongoose.model('test', parentSchema);

ParentModel.create({
	name: 'x',
	child: {
		name:'y'
	}
}, function(err, doc) {
	if(err) throw err;

	doc.child.name = 'z';
	doc.child.save(function(err, doc) {
		if(err) throw err;

		console.log(doc);
		mongoose.connection.close();
	});
});
```

**Comment by vkarpov15** [comment]
@EmailDE you're right, `doc.child.save()` is a noop. It's only there for middleware purposes.

**Comment by EmailDE** [comment]
Ah, ok. Though, other methods like `doc.child.$ignore()` also fail silently.

**Comment by vkarpov15** [comment]
@EmailDE you're right, that is definitely a problem as well. We'll fix.

**Comment by EmailDE** [comment]
Thats great! ;)
